### DISCUSSION TITLE
pupil preprocessing steps

### DISCUSSION THREAD

**Discussion by amjh23** [bachlab/PsPM#573] [Q&A] [answered]
Dear experts,

I am doing my first pupil analysis and would like to apply all preprocessing steps because my data are very noisy.
Is this the correct order to apply the preprocessing steps?: And does it make sense to use all these steps?
1) import
2) trim
3) pupil preprocessing (all defaults except for: increased maximum acceptable pupil size to 12mm)
4) convert gaze coordinates from pixel to length units
5) find valid fixations
6) PFE correction

This is the only order I could make work in pspm 6.1.0 using Eyelink1000 plus data. 
The reason for the difficulty is that both 3) and 6) produce a "pupil_pp" output channel, but both give error messages when "pupil_pp" is selected as input channel. Putting 5) in between solves the problem by creating again a "pupil" output channel. 

Many thanks for your help!

**Comment by dominikbach** [comment, marked as answer]
Thanks for posting this issue. 

I would suggest performing the pupil foreshortening error correction on the individual eyes, so if you wanted to combine them during the preprocessing that PFE correction should come first.

In the new release [6.1.1](https://github.com/bachlab/PsPM/releases/tag/v6.1.1), we have now changed the channel types and removed the type `pupil_pp`, such that all preprocessing steps generate the same channel type. You should be able to change the order now; if that does not work then please get in touch.
